database.go

```diff
@@ -21,29 +22,34 @@ func Connect(dial string) (*sql.DB, error) {
 	if dial == "" {
 		return nil, errors.New("empty database url provided")
 	}
-	url, err := url.Parse(dial)
+	parsedUrl, err := url.Parse(dial)
 	if err != nil {
 		return nil, err
 	}
 	// trim leading slash
-	dbname := strings.Trim(url.Path, "/")
-	fmt.Printf("Connecting to database '%s' on host '%s'\n", dbname, url.Host)
+	dbname := strings.Trim(parsedUrl.Path, "/")
+	fmt.Printf("Connecting to database '%s' on host '%s'\n", dbname, parsedUrl.Host)
 	return sql.Open("postgres", dial)
 }
 
 type Database interface {
-	Query(query string, args ...any) (*sql.Rows, error)
-	QueryRow(query string, args ...any) *sql.Row
-	Exec(query string, args ...any) (sql.Result, error)
+	ExecContext(context.Context, string, ...interface{}) (sql.Result, error)
```

Comment:
Providing context-aware methods seems fine, but I suggest we follow in the lead of the Go stdlib and provide them alongside the context-free versions rather than deleting those old versions. Though I don't think there are many non-nav users of this library, I'd rather not force a refactor on them.

Reply:
Oh I see, I was reading too quickly and thought you were changing the API that pomegranate gives to its users.  I see now you're just changing the interface that Pomegranate says it needs from the DB driver.  Looks good!